Source/MLX/MLXArray+Ops.swift

```diff
@@ -44,6 +44,29 @@ extension MLXArray {
         return MLXArray(mlx_add(lhs.ctx, rhs.ctx, s.ctx))
     }
 
+    /// Element-wise addition.
+    ///
+    /// Add two arrays with <doc:broadcasting> and store the result in the left-hand side array.
+    ///
+    /// For example:
+    ///
+    /// ```swift
+    /// let a = MLXArray(0 ..< 12, [4, 3])
+    /// let b = MLXArray([4, 5, 6])
+    ///
+    /// a += b
+    /// ```
+    ///
+    /// ### See Also
+    /// - <doc:arithmetic>
+    /// - ``add(_:_:stream:)``
+    public static func += (lhs: MLXArray, rhs: MLXArray) {
+        let s = StreamOrDevice.default
+        let result = mlx_add(lhs.ctx, rhs.ctx, s.ctx)!
+        mlx_free(lhs.ctx)
+        lhs.ctx = result
```

Comment:
Please use this function:

```swift
    /// Replace the contents with a reference to a new array.
    public func update(_ array: MLXArray) {
        if array.ctx != self.ctx {
            mlx_retain(array.ctx)
            mlx_free(ctx)
            self.ctx = array.ctx
        }
    }
```

Possibly make a package scoped variant that takes an `mlx_array` instead of `MLXArray`.  These self-mutates are rare and it is good to have them go through a common method.